tests/backup/backup_helper.go

```diff
@@ -1443,3 +1452,207 @@ func DeleteBackupAndWait(backupName string, ctx context.Context) error {
+	if err != nil {
+		return "", err
+	}
+	backupVersion := versionResponse.GetVersion()
+	return backupVersion.GetBuildDate(), nil
+}
+
+// UpgradePxBackup will perform the upgrade tasks for Px Backup to the version passed as string
+// Eg: versionToUpgrade := "2.4.0"
+func UpgradePxBackup(versionToUpgrade string) error {
+	var cmd string
+
+	// Compare and validate the upgrade path
+	currentBackupVersionString, err := GetPxBackupVersionSemVer()
+	if err != nil {
+		return err
+	}
+	currentBackupVersion, err := version.NewSemver(currentBackupVersionString)
+	if err != nil {
+		return err
+	}
+	versionToUpgradeSemVer, err := version.NewSemver(versionToUpgrade)
+	if err != nil {
+		return err
+	}
+
+	if currentBackupVersion.GreaterThanOrEqual(versionToUpgradeSemVer) {
+		return fmt.Errorf("px backup cannot be upgraded from version [%s] to version [%s]", currentBackupVersion.String(), versionToUpgradeSemVer.String())
+	} else {
+		log.InfoD("Upgrade path chosen (%s) ---> (%s)", currentBackupVersionString, versionToUpgrade)
+	}
+
+	// Getting Px Backup Namespace
+	pxBackupNamespace, err := backup.GetPxBackupNamespace()
+	if err != nil {
+		return err
+	}
+
+	// Delete the pxcentral-post-install-hook job is it exists
+	allJobs, err := batch.Instance().ListAllJobs(pxBackupNamespace, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	if len(allJobs.Items) > 0 {
+		log.Infof("List of all the jobs in Px Backup Namespace [%s] - ", pxBackupNamespace)
+		for _, job := range allJobs.Items {
+			log.Infof(job.Name)
+		}
+
+		for _, job := range allJobs.Items {
+			if strings.Contains(job.Name, pxCentralPostInstallHookJobName) {
+				err = deleteJobAndWait(job)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	} else {
+		log.Infof("%s job not found", pxCentralPostInstallHookJobName)
+	}
+
+	// Get storage class using for px-backup deployment
+	statefulSet, err := apps.Instance().GetStatefulSet(mongodbStatefulset, pxBackupNamespace)
+	if err != nil {
+		return err
+	}
+	pvcs, err := apps.Instance().GetPVCsForStatefulSet(statefulSet)
+	if err != nil {
+		return err
+	}
+	storageClassName := pvcs.Items[0].Spec.StorageClassName
+
+	// Get the tarball required for helm upgrade
+	cmd = fmt.Sprintf("curl -O  https://raw.githubusercontent.com/portworx/helm/%s/stable/px-central-%s.tgz", latestPxBackupHelmBranch, versionToUpgrade)
+	log.Infof("curl command to get tarball: %v ", cmd)
+	output, _, err := osutils.ExecShell(cmd)
+	if err != nil {
+		return fmt.Errorf("error downloading of tarball: %v", err)
+	}
+	log.Infof("Terminal output: %s", output)
+
+	// Execute helm upgrade using cmd
+	log.Infof("Upgrading Px-Backup version from %s to %s", currentBackupVersionString, versionToUpgrade)
+	cmd = fmt.Sprintf("helm upgrade px-central px-central-%s.tgz --namespace %s --create-namespace --version %s --set persistentStorage.enabled=true,persistentStorage.storageClassName=\"%s\",pxbackup.enabled=true",
```

Comment:
@mkoppal-px --create-namespace  should not be passed with upgrade command. --create-namespace will be used to create a namespace for fresh installation case.